lib/src/combined_wrappers/combined_iterator.dart

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// The iterator for `CombinedIterableView` and `CombinedListView`.
+///
+/// Moves through each iterable's iterator in sequence.
+class CombinedIterator<T> implements Iterator<T> {
+  /// The iterators that this combines, or `null` if done iterating.
+  ///
+  /// Because this comes from a call to [Iterable.map], it's lazy and will
```

Comment:
I worry this comment will become stale now that it is no longer an implementation detail of a single library. Since it should still be private to the package this might not be a huge deal, but we should consider moving the comment to the constructor to be something like "iterators should not eagerly instantiate iterators which have not been reached".

Reply:
It is still an implementation detail, the class is not exported by the package.